### PR TITLE
Fix all the docs links

### PR DESCRIPTION
**Pull request by davidpdrsn**
This goes through all the files and adds/fixes docs links. Also fixes typos and wordings in a few places.

### CODE REVIEW

File: tower-layer/src/identity.rs [hunk 1]
```diff
@@ -3,21 +3,25 @@ use std::fmt;
 
 /// A no-op middleware.
 ///
-/// When wrapping a `Service`, the `Identity` layer returns the provided
+/// When wrapping a [`Service`], the [`Identity`] layer returns the provided
```
Comment (hawkw) [inline tower-layer/src/identity.rs:6]: nit/TIOLI: does this type's docs _need_ to link to its own definition? you're already reading its docs...

**Review by hawkw — APPROVED** [review]
looks good to me! we should maybe consider adding a CI check for broken doc links to make this easier in the future...

File: tower-layer/src/identity.rs (hunk 1 above)
Comment (davidpdrsn) [reply tower-layer/src/identity.rs:6]: Yeah I did consider that as well however I don’t think it hurts and it’s just a single pair of `[]`. Also made it easier to search for everything inside backticks and make them into links.

**Comment by davidpdrsn** [comment]
I've added `#![deny(broken_intra_doc_links)]` to all crates now.

**Pull request merged**